### ISSUE TITLE
Mythic Sisters Lunar Beacon is often announced as being on the tank

### ISSUE THREAD

**Issue by cexikitin** [BigWigsMods/BigWigs#315]
https://vgy.me/1ADQye.png

https://www.warcraftlogs.com/reports/VKk8w3tp7na6XfcQ/#view=analytical&fight=16

I'll try and clip the actual combat log later.

**Comment by Zireael-N** [comment]
This is most likely because BW tries to predict the target during the cast by checking the target of the boss.

Even though it keeps rescanning for 0.5 seconds if its current target is the tank, I have found an example where she does not switch her target until the very end of the cast. Sorry for horrendous quality, I screen captured it from Twitch: https://gfycat.com/JampackedRaggedFlatfish

**Comment by cexikitin** [comment]
Maybe don't announce prediction if its on the tank? I think it'd be better to announce only once the target has been confirmed.

**Comment by Zireael-N** [comment]
I think the optimal workaround would be to do leave the prediction as is but if it returns a unit that's currently tanking the boss, fallback to announcing when the debuff is applied.

**Comment by Zireael-N** [comment]
If you happen to raid before a version with a fix released, here's a quick fix, open WoW/Interface/AddOns/BigWigs_TombOfSargeras/SistersoftheMoon.lua, delete the following line:
```Lua
self:GetBossTarget(printTarget, 0.5, args.sourceGUID) -- Faster than waiting for debuff/cast end
```
And replace this:
```Lua
function mod:LunarBeaconApplied(args)
	if self:Me(args.destGUID) then
		self:SayCountdown(args.spellId, 6)
	end
end
```
With this:
```Lua
function mod:LunarBeaconApplied(args)
	printTarget(self, args.destName, args.destGUID)
	if self:Me(args.destGUID) then
		self:SayCountdown(args.spellId, 6)
	end
end
```

**Comment by elvador** [comment]
I've increased the timer it waits in d64c5c7. This should fix the problem. If it still happens, please report back.

@Zireael-N Your quick fix wouldn't have worked because the `printTarget` func is defined after `mod:LunarBeaconApplied`.

**Comment by Zireael-N** [comment]
@elvador `bossScanner` seems to not allow itself to be called for longer than 0.8 seconds: https://github.com/BigWigsMods/BigWigs/blob/d64c5c768a17ef0353511cd21d70a73a61db3604/Core/BossPrototype.lua#L774

**Comment by elvador** [comment]
I know, but looking at videos I couldn't find any occurrences where it actually took longer than 0.5s, so the bump to 0.8s might be enough. If the problem still occurs we might need to change the value in BossPrototype.

**Comment by Justw8** [comment]
I think if it keep occuring, we'll have to prevent the targetscan from announcing the tank at all in this case (As it cannot go on tanks, unless they are the last ones standing) and just announce it once the debuff is actually applied as a fallback, because there wasn't a target found before that.

**Comment by Zireael-N** [comment]
I posted an example of the boss switching targets 1.4 seconds after the cast started in [this reply](https://github.com/BigWigsMods/BigWigs/issues/315#issuecomment-316268155). Personally, I don't think it is worth to keep scanning to get a notification 0.1s sooner, but I didn't actually profile anything. So, +1 to @Justw8's suggestion.

**Comment by Justw8** [comment]
it's worth to keep the scanning because there is the potential of a 1.5s warning before any debuff goes out. The late announcements are an exception, not the rule, from what I've seen.

**Comment by Zireael-N** [comment]
I meant scanning for up to 1.5 seconds, could've worded it better.

**Comment by elvador** [comment]
Didn't see your gif, fixed it with 93932d2.
